7.3/7.3/Program.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+namespace _7._3
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите позицию белого коня:");
+            var whiteKnightPosition = Console.ReadLine();
+
+            if (!IsPositionCorrect(whiteKnightPosition))
+            {
+                Console.WriteLine("Некорректная позиция белого коня");
+                Console.ReadKey();
+                return;
+            }
+
+            Console.WriteLine("Введите позицию черного слона:");
+            var blackBishopPosition = Console.ReadLine();
+
+            if (!IsPositionCorrect(blackBishopPosition) || whiteKnightPosition == blackBishopPosition)
+            {
+                Console.WriteLine("Черный слон не должен стоять на той же клетке, что и белый конь");
+                Console.ReadKey();
+                return;
+            }
+
+            if (IsBishopStrike(whiteKnightPosition, blackBishopPosition))
+            {
+                Console.WriteLine("Черный слон атакует белого коня");
+                Console.ReadKey();
+                return;
+            }
+            else
+            {
+                Console.WriteLine("Черный слон не атакует белого коня");
+                Console.WriteLine("Введите предполагаемый ход белого коня:");
+                var move = Console.ReadLine();
+
+                if (IsKnightMoveValid(whiteKnightPosition, move))
+                    Console.WriteLine("Ход разрешен");
```

Comment:
Нет проверки на то, что конь попадет под бой слона
```suggestion
                if (IsKnightMoveValid(whiteKnightPosition, move) &&
                        IsBishopStrike(move, blackBishopPosition))
                    Console.WriteLine("Ход разрешен");
```